qubessalt/__init__.py

```diff
@@ -108,31 +107,27 @@ def prepare_salt_config_for_vm(self):
         return output_dir
 
     def salt_call(self, command='state.highstate', return_output=False):
-        self.log.info('calling \'{}\'...'.format(command))
-        self.qc.lock_db_for_writing()
-        self.qc.load()
+        self.log.info('calling \'%s\'...', command)
         tpl = self.mgmt_template
 
         name = 'disp-mgmt-{}'.format(self.vm.name)
         # name is limited to 31 chars
         if len(name) > 31:
             name = name[:31]
-        dispvm = self.qc.get_vm_by_name(name)
+        dispvm = self.app.domains[name]
         if dispvm is None:
-            # FIXME: this should be DisposableVm type in core3
-            dispvm = self.qc.add_new_vm('QubesAppVm',
+            dispvm = self.app.add_new_vm('DispVm',
```

Comment:
Or rather:
```python
dispvm = qubes.vm.dispvm.DispVM.from_appvm(
    self.app.default_dispvm,
    netvm=None,
    internal=True)
```
TODO: consider adding `mgmt_dispvm` and use it here instead of `default_dispvm`